### ISSUE TITLE
nn.L1Loss takes input, target tensors that don't have broadcastable sizes

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#5007]
```
import torch
c = torch.nn.L1Loss()
a = torch.ones(10, 500, 80)
b = torch.zeros(10, 80, 500)
c(torch.autograd.Variable(a), torch.autograd.Variable(b))
```
per the documentation, this is not supposed to work. Either there is a bug or the documentation should be updated.

Originally reported here: https://discuss.pytorch.org/t/l1loss-returns-value-on-data-with-different-shape-but-different-number-of-elements/13161

**Comment by rafaelvalle** [comment]
Probably generalizes to other losses like L2Loss, etc...

**Comment by bheinzerling** [comment]
@rafaelvalle:

broadcastable does not mean "same number of elements", but roughly: "can be made the same size by repeating one of the tensors along exactly one axis".

The exact definition is given here:
http://pytorch.org/docs/master/notes/broadcasting.html

**Comment by zou3519** [comment]
The root issue is that L1Loss (along with some other losses) check if the input and target have the same number of elements. What it really should be doing is checking if the input and target have the same shapes. It looks like L1Loss (and most other losses, if not all) don't support broadcasting.
